src/commands/test_sonar.yml

```
@@ -1,24 +1,38 @@
+---
 description: "Setup steps"
 parameters:
   sonar_version:
     type: string
-    default: "3.0.3.778"
+    default: "4.2.0.1873"
   sonar_dir:
     type: string
     default: "/tmp/cache/sonar-scanner"
+  sonar_opts:
+    type: string
+    default: ""
   cache_version:
     type: integer
     default: 1
 steps:
+  - run:
+      name: Get SonarQube Server Version
```

```suggestion
      name: Generate sonar_version.txt
```
così è più chiaro l'intento

scambia i due restore cache o spoposta questo più sotto così è ancora più esplicito il quando viene usato